### DISCUSSION TITLE
Exclude screen variants and pseudo selectors for certain classes and specific values.

### DISCUSSION THREAD

**Discussion by neel1998** [tailwindlabs/tailwindcss#12817] [Help]
Is there a way to modify the config so that tailwind css doesn't generate screen and pseudo selector classes for certain values of a class.

For example I want sm: screen variant to be generated only for margin classes with values greater less than 3. 

so that i have sm:m-1, sm:m-2, sm:m-3 generated but not for any other margin classes.

**Comment by wongjn** [comment]
As long as you don't explicitly write those classes and don't have them in safelist, they shouldn't be generated. Otherwise, you could utilize the [`blocklist` configuration option](https://tailwindcss.com/docs/content-configuration#discarding-classes).

**Reply by neel1998** [reply]
hi thanks. but i don't want to rely on my content files and want to generate all the possible classes based on the config... however I am looking for some workaround to exclude certain pseudo selectors for specific classes.

**Reply by wongjn** [reply]
You could utilize the [`blocklist` configuration option](https://tailwindcss.com/docs/content-configuration#discarding-classes) to exclude certain classes.

**Reply by neel1998** [reply]
yes, but unfortunately blocklist doesn't allow for regex so i'll have to give all the specific classnames

**Reply by wongjn** [reply]
Indeed. You can programmatically generate the block list to at least not to have to write each one out. Alternatively, if you are using Tailwind as a PostCSS plugin, you could write a PostCSS plugin that runs after Tailwind that removes the undesired classes from the CSS.

**Reply by neel1998** [reply]
yup i thought the same. Thanks!
